### PR TITLE
feat: support no_std

### PR DESCRIPTION
**Pull request by yjhmelody**

### CODE REVIEW

**Comment by yjhmelody** [comment]
Hi @DaniPopes @rkrasiuk  PTAL. Any problem?

File: Cargo.toml [hunk 1]
```diff
@@ -20,15 +20,15 @@ repository = "https://github.com/alloy-rs/trie"
...
+alloy-primitives = { version = "0.6", default-features = false, features = ["rlp"] }
+alloy-rlp = { version = "0.3", default-features = false, features = ["derive"] }
 derive_more = "0.99"
-nybbles = "0.1"
+nybbles = { version = "0.1", default-features = false }
 smallvec = "1.11"
-tracing = "0.1"
-
+tracing = { version = "0.1", default-features = false }
+hashbrown = { version = "0.14", features = ["serde", "ahash", "inline-more"] }
```
Comment (DaniPopes) [inline Cargo.toml]: serde feature has to be enabled in the `[features]` section

File: Cargo.toml [hunk 2]
```diff
@@ -20,15 +20,15 @@ repository = "https://github.com/alloy-rs/trie"
...
 # serde
```
Comment (DaniPopes) [inline Cargo.toml:31]:
```suggestion

# serde
```

File: src/lib.rs [hunk 3]
```diff
@@ -14,6 +14,10 @@
 #![cfg_attr(not(test), warn(unused_crate_dependencies))]
 #![deny(unused_must_use, rust_2018_idioms)]
 #![cfg_attr(docsrs, feature(doc_cfg, doc_auto_cfg))]
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg(not(feature = "std"))]
+extern crate alloc as std;
```
Comment (DaniPopes) [inline src/lib.rs]: Can you remove the alias? I prefer being explicit about `alloc`, plus it doesn't re-export items from `core` like `std` does.

File: src/lib.rs [hunk 4]
```diff
@@ -24,6 +28,8 @@ pub use hash_builder::HashBuilder;
 mod mask;
 pub use mask::TrieMask;
 
+pub use hashbrown::HashMap;
```
Comment (DaniPopes) [inline src/lib.rs:32]: Should we use `std` hashmap when `std` feature is on?

**Review by DaniPopes — CHANGES_REQUESTED** [review]

**Comment by yjhmelody** [comment]
done

**Comment by yjhmelody** [comment]
@DaniPopes

**Review by DaniPopes — APPROVED** [review]

**Comment by DaniPopes** [comment]
Released in [0.2.1](https://crates.io/crates/alloy-trie/0.2.1)

**Pull request merged**